sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -95,20 +95,16 @@ private static void CheckCountCall(SyntaxNodeAnalysisContext context)
             const string CountName = "Count";
 
             var invocation = (InvocationExpressionSyntax)context.Node;
-            if (!(context.SemanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol) ||
-                methodSymbol.Name != CountName ||
-                invocation.ArgumentList == null ||
+            if (invocation.ArgumentList == null ||
```

Comment:
I would try to simplify this a bit. Maybe something similar with:
```
var invocation = (InvocationExpressionSyntax)context.Node;
if (invocation.ArgumentList?.Arguments.Count == 0 &&
    invocation.Expression is MemberAccessExpressionSyntax memberAccess &&
    context.SemanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol &&
    methodSymbol.Name == CountName &&
    methodSymbol.IsExtensionOn(KnownType.System_Collections_Generic_IEnumerable_T) &&
    context.SemanticModel.GetTypeInfo(memberAccess.Expression).Type.GetMembers(CountName).OfType<IPropertySymbol>().Any())
{
    context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, GetReportLocation(invocation),
                                                         string.Format(MessageUseInstead, $"'{CountName}' property")));
}
```

and maybe extract `context.SemanticModel.GetTypeInfo(memberAccess.Expression).Type.GetMembers(CountName).OfType<IPropertySymbol>().Any()` to a method (can be local)

In this way we have a single list with preconditions and avoid all the negated checks. It is also more concise.